sources/libs/hw/fdt/fdt.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <brutal/base.h>
+#include <brutal/base/endian.h>
+#include <brutal/io/emit.h>
+
+typedef struct PACKED
+{
+    be_uint32_t magic;
+    be_uint32_t size;
+    be_uint32_t structure_offset;
+    be_uint32_t strings_offset;
+    be_uint32_t memory_reservation_offset;
+    be_uint32_t version;
+    be_uint32_t last_comp_version;
+    be_uint32_t boot_cpuid_phys;
+    be_uint32_t strings_size;
+    be_uint32_t structure_size;
+} FdtHeader;
+
+typedef struct PACKED
+{
+    be_uint64_t address;
+    be_uint64_t size;
+} FdtReservationEntry;
+
+typedef enum
+{
+    FDT_BEGIN_NODE = 1,
+    FDT_END_NODE = 2,
+    FDT_PROP = 3,
+    FDT_NOP = 4,
+    FDT_END = 9
+} FdtNodeType;
+
+typedef struct PACKED ALIGNED(4)
+{
+    be_uint32_t type;
+    char name[];
+} FdtBeginNode;
+
+typedef struct PACKED
+{
+    be_uint32_t type;
+    be_uint32_t len;
+    be_uint32_t name_offset;
+    char value[];
+} FdtPropertiesNode;
+
+typedef struct
+{
+    Str property_name;
+    size_t property_value_size;
+    void *property_value;
+} NodeProperty;
```

Comment:
The property_* prefix is not useful, we already know it's a property from the type name 

Also, it's missing the Fdt prefix